### PR TITLE
Support for streaming/downloading files

### PR DESCRIPTION
**Pull request by shasha79**
Added support and test for streaming/downloading files API using stream=True parameter for session.request() method (its usage is described here: http://docs.python-requests.org/en/master/user/advanced/#body-content-workflow)

### CODE REVIEW

File: tests/integration/test_stream.tavern.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+
+test_name: Test streaming (downloading) file
+
+includes:
+  - !include common.yaml
+
+stages:
+  - name: Stream file
```
Comment (michaelboulton) [inline tests/integration/test_stream.tavern.yaml]: Should this stage have the 'stream' flag?

File: tavern/_plugins/rest/request.py [hunk 2]
```diff
@@ -86,7 +87,7 @@ def get_request_args(rspec, test_block_config):
             rspec["headers"] = {i: j for i, j in headers.items() if i.lower() != "content-type"}
     elif headers:
         # This should only be hit if we aren't sending a file
-        if not has_content_header:
+        if not has_content_header and not rspec.get("stream"):
```
Comment (michaelboulton) [inline tavern/_plugins/rest/request.py]: I don't understand why this check is here - the streaming should affect the response, not the request content-type header

**Comment by michaelboulton** [comment]
Thanks for that, It should be deployed in version 0.14.5

**Comment by shasha79** [comment]
Thanks for the great library, happy to contribute!

**Pull request merged**